jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/client/ClientTracingFeature.java

```diff
@@ -0,0 +1,106 @@
+package io.opentracing.contrib.jaxrs.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.ws.rs.client.Client;
+
+import io.opentracing.Tracer;
+
+/**
+ * @author Pavol Loffay
+ */
+public class ClientTracingFeature {
+
+    private static final Logger log = Logger.getLogger(ClientTracingFeature.class.getName());
+
+    private ClientTracingFeature(Builder builder) {
+        if (log.isLoggable(Level.INFO)) {
+            log.info("Registering client tracing for: " + builder.client);
+        }
+
+        builder.client.register(new SpanClientRequestFilter(builder.tracer, builder.spanDecorators), 0)
+                .register(new SpanClientResponseFilter(builder.spanDecorators), 0);
+    }
+
+    /**
+     * Builder for configuring {@link Client} to trace outgoing requests.
+     *
+     * By default span operation name is set by {@link ClientSpanDecorator#HTTP_METHOD_OPERATION_NAME} and
+     * span is decorated with {@link ClientSpanDecorator#STANDARD_TAGS}.
+     */
+    public static class Builder {
+        private Tracer tracer;
+        private Client client;
+        private List<ClientSpanDecorator> spanDecorators = new ArrayList<>();
```

Comment:
note: since you are exposing a list, basically you can expect people to complain about order and other types of problems related to that. Usually I don't, and instead punt to the user to compose their own decorator lists.